### ISSUE TITLE
503 returned when calling produce

### ISSUE THREAD

**Issue by jcarroll-mediafly** [fastlane/fastlane#10812]
### New Issue Checklist

- [√] Updated fastlane to the latest version
- [√] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [√] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [√] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Unsure if this is an apple issue or fastlane. But I'm constantly getting 503 when using `produce`. Had my team members run the lane as well and same issue. I found an older ticket https://github.com/fastlane/fastlane/issues/8209 with similar problems, and it seems unclear if its an apple thing or fastlane even. Of course apple services are all green at the moment.

I will say we have ~50 app ids and periodically we run this lane to ensure the extensions for each app id has the correct app id, services enabled etc turned on / configured correctly. Basically its a lane to ensure dev portal is up to snuff without having to click 5000 times or make a human mistake of reading or changing something incorrectly.

Anyone else have 503 issues? Our latest guess is apple is possibly blocking / rate limiting us.

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.3                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.15.0                          |
| Installation Source         | ~/.rvm/gems/ruby-2.2.3/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G1036)                  |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.2.3/lib                |
| OpenSSL Version             | OpenSSL 1.0.2h  3 May 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.63.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.2        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.6       |
| faraday_middleware        | 0.10.1       |
| declarative               | 0.0.8        |
| representable             | 3.0.2        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.2       |
| json                      | 2.0.2        |
| rubyzip                   | 1.2.0        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
</details>

*generated on:* **2017-11-06**
</details>

**Comment by yosi-dediashvili** [comment]
Same here.

Spaceship is getting `Spaceship::TunesClient::ITunesConnectError` with `503 Service Temporarily Unavailable`

**Comment by mpatzer** [comment]
Also getting a 503 when trying to `fastlane match`, including `fastlane match nuke`. Started in the past hour.

**Comment by mpatzer** [comment]
Working now this morning. No more 503.

**Comment by mpatzer** [comment]
And a 503 again. So my best guess is that we're being rate-limited by Apple. We have 9 app id's that we're using match to generate provisioning profiles for in development, adhoc, and distribution configurations.

**Comment by jcarroll-mediafly** [comment]
Guess its time to make 1000 clicks in portal :/

**Comment by mpatzer** [comment]
I tried waiting 1 hour before attempting the call again, since that seems like a reasonable cool-down period, but no luck, got a `503` again. The cool-down period for rate limiting must be longer than 1 hour.

**Comment by taquitos** [comment]
Thanks for the report, we're checking this out, seems like the same root cause as https://github.com/fastlane/fastlane/issues/10793

**Comment by taquitos** [comment]
I think this PR should fix it: https://github.com/fastlane/fastlane/pull/10826

**Comment by taquitos** [comment]
I wanted to update folks around the auth issues we’re seeing right now.
Context:
Issue first seen in https://github.com/fastlane/fastlane/issues/10793
Picked up again in https://github.com/fastlane/fastlane/issues/10812

While I have a PR up that fixes this for the simple login case, it only addresses SMS 2FA, and it employs web scraping, which is super flakey. 

Furthermore, the authentication logic we already have addresses so many corner cases, and is battle tested. Replacing that with this flakey update could cause a lot of pain for everybody.

Instead, we think the right thing moving forward is to assume Apple isn’t going to fix this obvious issue, and then focus our energy on improving our session caching, so that between various actions, we don’t login multiple times unless explicitly called for.

This will take some time to properly address, but we’ll have a plan soon and we’ll share that out as soon as we do 🚀. In the meantime, we’re going to keep my PR (https://github.com/fastlane/fastlane/pull/10826) alive if it is helping to unblock people.

Repeating what I posted before, we have some workarounds whose success seem to vary:

- Login using a different account from same IP
- Change IP address (VPN/Proxy)
- Login from a different IP, it unblocks your original IP.
- Login in developer.apple.com and then navigate to itunesconnect.apple.com

**Comment by KrauseFx** [comment]
Hey everyone, thanks again for looking into this issue. Sorry it took so long, I finally found some time to look further into this issue, and was able to reproduce the issue and prepare a fix. You can check out the PR over here https://github.com/fastlane/fastlane/pull/11108

I'd love to ship this as a new release tomorrow morning 👍
